WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift

```diff
@@ -112,17 +119,47 @@ final class BlazeConfirmPaymentViewModel: ObservableObject {
         }
 
         analytics.track(event: .Blaze.Payment.submitCampaignTapped())
-        shouldDisplayCampaignCreationError = false
+        campaignCreationError = nil
         isCreatingCampaign = true
         do {
-            let updatedDetails = campaignInfo.copy(paymentMethodID: selectedPaymentMethod.id)
-            try await requestCampaignCreation(details: updatedDetails)
+            // Prepare image for campaign
+            let campaignMedia: Media
+            switch image.source {
+            case .asset(let asset):
+                do {
+                    campaignMedia = try await uploadPendingImage(asset)
+                } catch {
+                    DDLogError("⛔️ Error uploading campaign image: \(error)")
+                    throw BlazeCampaignCreationError.failedToUploadCampaignImage
+                }
+            case .media(let media):
+                campaignMedia = media
+            case .productImage(let image):
+                do {
+                    campaignMedia = try await retrieveMedia(mediaID: image.imageID)
+                } catch {
+                    DDLogError("⛔️ Error fetching product image's Media: \(error)")
+                    throw BlazeCampaignCreationError.failedToFetchCampaignImage
+                }
+            }
```

Comment:
Suggestion: should we move this to a separate method for readability?

```suggestion
            let campaignMedia = prepareImage(from: image.source)
```

Reply:
Good suggestion. Done in https://github.com/woocommerce/woocommerce-ios/pull/12006/commits/59d959ce08bc28846d063f7c25c502368cecd6b8